src/framework/components/session/core.clj

```diff
@@ -0,0 +1,23 @@
+(ns framework.components.session.core)
+
+(defprotocol Session
+  (fetch
+    [store key])
+  (add
+    [store key value])
+  (delete
+    [store key]))
+
+(deftype MemorySession [store]
+         Session
+         (fetch [_ k]
+           (get @store k))
+         (add [_ k v]
```

Comment:
these ones has mutability on their impl, it should be good to have `!` signature